Mas12517/realtime weather (#72)
Esri/arcgis-maps-sdk-unreal-engine-samples: sample_project/Content/SampleViewer/Samples/RealTime_Weather/readme.md

> # Query a feature layer

Reviewer: ```suggestion
# Real-time weather
```

> Query objects from the feature layer provided, or provide your own, in order to show them on the map.

Reviewer: the description needs to be updated

Author: Technically this description isn't wrong since it is a feature layer XD
I'll update it though :)

![Image of Weather Sample](Weather.png)

## How to Setup (Weather Sample Level)

1. Open the **RealTimeWeather** level.
2. Click on the **ArcGISMapActor** in the Outliner panel.
3. Set your API key under the Authentication section in the Details panel.
4. Click Play and select a city from the list.

## How to Setup (Sample Viewer)

1. Click Play in Unreal Editor.
2. Input your API key under the **API Key Drop-down**.
3. Click the **Sample Drop-Down** and select **Weather Query**.

## How it works

1. Create a new C++ class and make a http request to [query a feature layer](https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-.htm). 
2. Create a new Blueprint Actor class
   - Create the event to place the data returned from the Weather Query.

>    - Create a function to spawn the actor according to the data received  in the query.

Reviewer 2: ```suggestion
   - Create a function to spawn the weather actor according to the data received in the query.
```

   - Attach the [**ArcGIS Location Component**](https://developers.arcgis.com/unreal-engine/maps/location-component/) to the weather actor blueprint class.
   - Add aditional functionality to spawn/show different **Niagra Particle Effects** based on the current weather data returned from the **Weather Query**.

> 3. Create a widget for the viewport so users can select a city from the list fed from the feature service.

Reviewer 2: ```suggestion
3. Create a widget for the viewport so users can select an item from the list fed from the feature service response.
```

Note: You can use `UE_LOG` to print log messages in the **Output Log** window and see if you are gathering the data properly from the feature service.

## About the data

> Data for Current Weather in the United States [Feature Layer](https://services9.arcgis.com/RHVPKKiFTONKtxq3/ArcGIS/rest/services/NOAA_METAR_current_wind_speed_direction_v1/FeatureServer/0//query?where=COUNTRY+LIKE+%27%25United+States+of+America%27+AND+WEATHER+NOT+IN(%27No+significant+weather+present+at+this+time.%27%2C+%27Automated+observation+with+no+human+augmentation%3B+there+may+or+may+not+be+significant+weather+present+at+this+time.%27)&outFields=*&f=pgeojson) hosted by Esri.

Reviewer 2: ```suggestion
Data for Current Weather in the United States is fetched from a [Feature Layer](https://services9.arcgis.com/RHVPKKiFTONKtxq3/ArcGIS/rest/services/NOAA_METAR_current_wind_speed_direction_v1/FeatureServer/0//query?where=COUNTRY+LIKE+%27%25United+States+of+America%27+AND+WEATHER+NOT+IN(%27No+significant+weather+present+at+this+time.%27%2C+%27Automated+observation+with+no+human+augmentation%3B+there+may+or+may+not+be+significant+weather+present+at+this+time.%27)&outFields=*&f=pgeojson) hosted by Esri.
```

# Query a feature layer

Query objects from the feature layer provided, or provide your own, in order to show them on the map.

![Image of Weather Sample](Weather.png)

## How to Setup (Weather Sample Level)

1. Open the **RealTimeWeather** level.
2. Click on the **ArcGISMapActor** in the Outliner panel.

> 3. Set your API key under Authentication section in the Details panel.

Reviewer: ```suggestion
3. Set your API key under the Authentication section in the Details panel.
```

> 4. Click play and select a city from the list.

Reviewer: ```suggestion
4. Click Play and select a city from the list.
```

## How to Setup (Sample Viewer)

1. Click Play in Unreal Editor.

> 2. Input your API key under the **API Key Drop down**.

Reviewer: ```suggestion
2. Input your API key under the **API Key Drop-down**.
```

> 3. Click the **Sample Drop Down** and select **Weather Query**.

Reviewer: ```suggestion
3. Click the **Sample Drop-down** and select **Weather Query**.
```

## How it works

1. Create a new C++ class and make a http request to [query a feature layer](https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-.htm). 
2. Create a new Blueprint Actor class
   - Create the event to place the data returned from the feature layer.

>    - Create a function to spawn the actor according to the data recieved in teh query.

Reviewer: ```suggestion
   - Create a function to spawn the actor according to the data received in the query.
```

   - Attach the [**ArcGIS Location Component**](https://developers.arcgis.com/unreal-engine/maps/location-component/) to the static stadium mesh model.
   - Create a function to place the static mesh model on the `geometry`'s `coordinates` location returned by the feature layer query and the height calculated by [raycasts](https://docs.unrealengine.com/5.0/en-US/using-a-single-line-trace-raycast-by-channel-in-unreal-engine/).

> 3. Create a widget for the viewport so users can select a stadium from the list fed from the feature service.

Reviewer: still says stadium

Note: You can use `UE_LOG` to print log messages in the **Output Log** window and see if you are gathering the data properly from the feature service.

## About the data

> Data for all Major League Baseball Stadiums in the United States [Feature Layer](https://www.arcgis.com/home/item.html?id=f60004d3037e42ad93cb03b9590cafec) hosted by Esri. (Format the request URL for the [query](https://services.arcgis.com/P3ePLMYs2RVChkJx/ArcGIS/rest/services/Major_League_Baseball_Stadiums/FeatureServer/0/query?f=geojson&where=1=1&outfields=TEAM,NAME,LEAGUE).)

Reviewer: Shouldn't this be the weather data?
